tools/model.py

```diff
@@ -257,7 +351,54 @@ def data(
             return dfs
 
     def shares(self):
-        return self.client._shares(resource=self)
+        """
+        Retrieve list of shares
+
+        :return: list of ResourceShares
+        """
+        return self.client._list_shares(resource_id=self.id)
+
+    def share(
+        self, user_email: str, permission_level: PermissionLevel = PermissionLevel.view
+    ):
+        """
+        Share the Synthetic Data with specified user and access rights (default PermissionLevel.view)
+        """
+        self.client._share(
+            resource_id=self.id,
+            user_email=user_email,
+            permission_level=permission_level,
+        )
+
+    def unshare(self, user_email: str):
+        """
+        Stop sharing the Synthetic Data with the specified user
```

Comment:
use `synthetic dataset` consistently instead of `Synthetic Data` throughout the code